### PR TITLE
ordering transfers

### PR DESCRIPTION
**Pull request by matei-p**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -36,3 +36,5 @@ require (
 	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+// replace github.com/multiversx/mx-chain-vm-common-go v1.4.2-0.20230425081117-bd6069791f3b => /home/bogdan/Elrond/elrond-vm-common
```
Comment (sasurobert) [inline go.mod:40]: delete

File: testcommon/vmOutputVerifier.go [hunk 2]
```diff
@@ -314,10 +314,20 @@ type TransferEntry struct {
...
 // CreateTransferEntry creates the data for an output transfer assertion
-func CreateTransferEntry(senderAddress []byte, receiverAddress []byte) *TransferEntry {
+// func CreateTransferEntry(senderAddress []byte, receiverAddress []byte) *TransferEntry {
+// 	return &TransferEntry{
+// 		OutputTransfer: vmcommon.OutputTransfer{SenderAddress: senderAddress},
+// 		address:        receiverAddress,
+// 	}
+// }
+
+// TODO RENAME THIS from CreateTransferEntryWithIndex to CreateTransferEntry
```
Comment (sasurobert) [inline testcommon/vmOutputVerifier.go:326]: resolve todo's

File: vmhost/contexts/asyncParams.go [hunk 3]
```diff
@@ -16,6 +17,7 @@ import (
     TODO(fix) this function
```
Comment (sasurobert) [inline vmhost/contexts/asyncParams.go:17]: can we delete this todo ?

File: vmhost/contexts/output.go [hunk 4]
```diff
@@ -33,8 +34,9 @@ func NewOutputContext(host vmhost.VMHost) (*outputContext, error) {
 	}
 
 	context := &outputContext{
-		host:       host,
-		stateStack: make([]*vmcommon.VMOutput, 0),
+		host:             host,
+		stateStack:       make([]*vmcommon.VMOutput, 0),
+		crtTransferIndex: 0,
```
Comment (sasurobert) [inline vmhost/contexts/output.go:39]: you need to reset the crtTransferIndex on InitState. That is called only once per transaction.

Comment (matei-p) [reply vmhost/contexts/output.go:39]: good catch

File: mock/contracts/esdtSimpleSC.go [hunk 5]
```diff
@@ -37,8 +37,9 @@ func ExecESDTTransferAndCallChild(instanceMock *mock.InstanceMock, config interf
 		input.Arguments = [][]byte{
 			test.ESDTTestTokenName,
 			big.NewInt(int64(testConfig.ESDTTokensToTransfer)).Bytes(),
-			arguments[2],
+			// arguments[2],
```
Comment (sasurobert) [inline mock/contracts/esdtSimpleSC.go:40]: delete

File: mock/context/outputContextMock.go [hunk 6]
```diff
@@ -206,3 +206,18 @@ func (o *OutputContextMock) CreateVMOutputInCaseOfError(_ error) *vmcommon.VMOut
 func (o *OutputContextMock) GetCurrentTotalUsedGas() (uint64, bool) {
 	return 0, false
 }
+
+// NextOutputTransferIndex mocked method
+func (o *OutputContextMock) NextOutputTransferIndex() uint32 {
+	return 0
```
Comment (sasurobert) [inline mock/context/outputContextMock.go:212]: with the new code this might fail. Internal incrementing in this Mock.

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**